src/image_builder.py

```diff
@@ -154,109 +156,198 @@ def image_builder(buildspec):
             "labels": labels,
             "extra_build_args": extra_build_args
         }
-
-        image_object = DockerImage(
+        
+        #Create initial stage docker object
+        initial_stage_image_object = DockerImage(
             info=info,
             dockerfile=image_config["docker_file"],
             repository=image_repo_uri,
             tag=image_tag,
             to_build=image_config["build"],
+            stage=constants.INITIAL_STAGE,
```

Comment:
Should be `PREVIOUS_STAGE` as mentioned earlier

Reply:
Done.